client/my-sites/jetpack-search/placeholder.tsx

```diff
@@ -20,12 +24,17 @@ import './style.scss';
 
 interface Props {
 	siteId: number;
+	isJetpack: boolean;
 }
 
-export default function JetpackSearchPlaceholder( { siteId }: Props ): ReactElement {
+export default function JetpackSearchPlaceholder( { siteId, isJetpack }: Props ): ReactElement {
+	const currentUserId = useSelector( getCurrentUserId );
 	return (
 		<Main className="jetpack-search__placeholder">
-			<QuerySitePurchases siteId={ siteId } />
+			<QueryUserPurchases userId={ currentUserId } />
```

Comment:
Querying all user purchases here to avoid being without site purchases during a site switch. The request is a bit heavier but helps prevent some undesirable behaviour (like a flash of the upsell during the site switch while we wait for the purchases).